source/common/tracing/http_tracer_impl.cc

```diff
@@ -210,7 +210,9 @@ void HttpTracerUtility::finalizeDownstreamSpan(Span& span,
   span.setTag(Tracing::Tags::get().RequestSize, std::to_string(stream_info.bytesReceived()));
   span.setTag(Tracing::Tags::get().ResponseSize, std::to_string(stream_info.bytesSent()));
 
-  setCommonTags(span, response_headers, response_trailers, stream_info, tracing_config);
+  setCommonTags(span, stream_info, tracing_config);
+  onUpstreamResponseHeaders(span, response_headers);
```

Comment:
should these still be called here?  is there a chance that gRPC tags will be added twice or is that safe?

Reply:
> should these still be called here?

Yes. Envoy supports gRPC downstream and currently sets the gRPC tags for the "main" downstream span. This preserves the behavior.

> is there a chance that gRPC tags will be added twice or is that safe?

Should be safe. gRPC spec guarantees `grpc-status` only occurs once. https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md

Reply:
Awesome, that's what I needed to know, thanks!